python/ml4ir/base/model/relevance_model.py

```diff
@@ -810,3 +813,44 @@ def calibrate(self, relevance_dataset, logger, logs_dir_local, temperature_init)
                                  logs_dir_local=logs_dir_local,
                                  temperature_init=temperature_init,
                                  file_io=self.file_io)
+
+    def add_temperature_layer(self, temperature: float):
+        """Add temperature layer to the input of last activation (softmax) layer
+        Parameters
+        ----------
+        self: RelevanceModel
+                        input RelevanceModel object that its last layer inputs will be divided by a
+                        temperature value
+        temperature: float
+                     a scalar value to scale the last activation layer inputs
+        Returns
+        -------
+        `RelevanceModel`
+        updated RelevanceModel object with temperature
+        """
+
+        # get  last layer's output  --> MUST **NOT** BE AN ACTIVATION (e.g. SOFTMAX) LAYER
+        final_layer_name = self.scorer.model_config['layers'][-1]['name']
+
+        final_layer = self.model.get_layer(name=final_layer_name).output
+        temperature_layer = TemperatureScalingLayer(name='temperature_layer',
+                                                    temperature=temperature)(final_layer)
+
+        # using the `last Activation layer` as final activation function before computing loss
+        i = 1
```

Comment:
you should use a more descriptive variable name here, what you are going to do in the next lines is not trivial.